objcgen/embedder.cs

```diff
@@ -438,6 +438,7 @@ public int Compile ()
 							options.Append ($"-dynamiclib ");
 							options.Append (build_info.LinkerFlags).Append (" ");
 							options.Append ("-lobjc ");
+							options.Append ("-lc++ ");
```

Comment:
It's not clear reading the diff (and it's not part of the PR description) if this requirement is **only** when linking with XI/XM or if it also affects a native ObjC application that wants to use a normal .dll assembly.

My understanding is that the later scenario should *not* require linking with C++ libraries and, for various reasons, should be avoided.

We can discuss separately if XI/XM integration requires (or not) linking with C++.